src/lib/flow/sol-flow-builder.c

```diff
@@ -116,6 +117,9 @@ sol_flow_builder_del(struct sol_flow_builder *builder)
     }
     sol_ptr_vector_clear(&builder->ports_out_desc);
 
+    free((char *)builder->type_desc.symbol);
+    free((char *)builder->type_desc.options_symbol);
```

Comment:
isn't it better to use a sol_arena to store those and cleanup the whole thing at the end?